classes/pdf/PDF.php

```diff
@@ -192,4 +187,40 @@ public function getTemplateObject($object)
 
         return $class;
     }
+
+    /**
+     * Get the PDF filename based on the objects.
+     *
+     * @return string
+     */
+    public function getFilename(): string
+    {
+        if (empty($this->filename)) {
+            $this->setFilename();
+        }
+
+        return $this->filename;
+    }
+
+    /**
+     * Set the PDF filename based on the objects.
+     *
+     * @return bool
+     */
+    public function setFilename(): bool
+    {
+        foreach ($this->objects as $object) {
+            $template = $this->getTemplateObject($object);
+            if (!$template) {
+                continue;
+            }
+
+            $this->filename = $template->getFilename();
+            if (count($this->objects) > 1) {
+                $this->filename = $template->getBulkFilename();
+            }
+        }
+
+        return !empty($this->filename);
```

Comment:
@JoryHogeveen I think @sowbiba is right, the loop misses a break or a return as soon as the name is defined.
```suggestion
            }
            if (!empty($this->filename)) {
                return true;
            }
        }

        return false;
```

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/26587/commits/a6b421b9630f79353d29d67ae6d7ef96eb5a268b